### PR TITLE
feat: show warning in pr body

### LINKED ISSUE #16924: Show warnings indication in non-onboarding PR bodies
### What would you like Renovate to be able to do?

Show an indication in each non-onboarding PR if there were warnings encountered

### If you have any ideas on how this should be implemented, please tell us here.

I don't think we need to list all the warnings, instead we could "warn if there's warnings" and tell people to check the dependency dashboard or logs. Ideally we would say dashboard if it's enabled or logs if it's not.

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by MaronHatoum**
## Changes
1. warnings has been added to PR body
2. display a relevant warnings msg (dependencyDashboard=true/false) in pr body

## Context
closes https://github.com/renovatebot/renovate/issues/16924 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by MaronHatoum** [comment]
Examples from real repository:

```
{
  "dependencyDashboard": true
}
```
https://github.com/MaronHatoum/warnings-in-PR-bodies/pull/55


```
{
  "dependencyDashboard": false
}
```
https://github.com/MaronHatoum/warnings-in-PR-bodies/pull/56

File: lib/workers/repository/errors-warnings.ts [hunk 1]
```diff
@@ -69,14 +70,11 @@ export function getDepWarningsPR(
   warningText = emojify(
     `\n---\n\n### :warning: Dependency Lookup Warnings :warning:\n\n`
   );
-  warningText += `Please correct - or verify that you can safely ignore - these lookup failures before you merge this PR.\n\n`;
-  for (const w of warnings) {
-    warningText += `-   \`${w}\`\n`;
+  if (dependencyDashboard) {
+    warningText += `Warnings were logged while processing this repo. Please check the Dependency Dashboard for further information\n\n`;
+  } else {
+    warningText += `Warnings were logged while processing this repo. Please check logs for further information\n\n`;
```
Comment (nabeelsaabna) [inline lib/workers/repository/errors-warnings.ts:76]:
```suggestion
  warningText += `Warnings were logged while processing this repo.`;
  if (dependencyDashboard) {
    warningText += ` Please check the Dependency Dashboard for further information\n\n`;
  } else {
    warningText += ` Please check logs for further information\n\n`;
```

File: lib/workers/repository/errors-warnings.ts [hunk 2]
```diff
@@ -69,14 +70,11 @@ export function getDepWarningsPR(
   warningText = emojify(
     `\n---\n\n### :warning: Dependency Lookup Warnings :warning:\n\n`
   );
-  warningText += `Please correct - or verify that you can safely ignore - these lookup failures before you merge this PR.\n\n`;
-  for (const w of warnings) {
-    warningText += `-   \`${w}\`\n`;
+  if (dependencyDashboard) {
```
Comment (HonkingGoose) [inline lib/workers/repository/errors-warnings.ts:73]: Can you change the  `warningText` strings, so the final result looks like this on the PR body?

> Warnings were logged while processing this repo. Please check the Dependency Dashboard for more information.

> Please check the logs for more information.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/workers/repository/errors-warnings.ts (hunk 2 above)
Comment (MaronHatoum) [reply lib/workers/repository/errors-warnings.ts:74]: @betterPeleg

File: lib/workers/repository/update/pr/body/index.ts [hunk 3]
```diff
@@ -69,9 +70,18 @@ export async function getPrBody(
   prBodyConfig: PrBodyConfig
 ): Promise<string> {
   massageUpdateMetadata(branchConfig);
+  let warn = '';
+  warn += getWarnings(branchConfig);
+  if (branchConfig.packageFiles) {
+    warn += getDepWarningsPR(
```
Comment (Gabriel-Ladzaretti) [inline lib/workers/repository/update/pr/body/index.ts:76]:
```suggestion
  let warnings = '';
  warnings += getWarnings(branchConfig);
  if (branchConfig.packageFiles) {
    warnings += getDepWarningsPR(
```

File: lib/workers/repository/update/pr/body/index.ts [hunk 4]
```diff
@@ -69,9 +70,18 @@ export async function getPrBody(
...
+      branchConfig.packageFiles,
+      branchConfig.dependencyDashboard
+    );
+  }
   const content = {
     header: getPrHeader(branchConfig),
     table: getPrUpdatesTable(branchConfig),
+    warnings: warn,
```
Comment (Gabriel-Ladzaretti) [inline lib/workers/repository/update/pr/body/index.ts:84]:
```suggestion
    table: getPrUpdatesTable(branchConfig),
    warnings,
```

**Review by Gabriel-Ladzaretti — COMMENTED** [review]
Nit

File: lib/workers/repository/errors-warnings.spec.ts [hunk 5]
```diff
@@ -82,19 +83,61 @@ describe('workers/repository/errors-warnings', () => {
...
+      };
+
+      const res = getDepWarningsPR(packageFiles, dependencyDashboard);
+      expect(res).toMatchInlineSnapshot(`
+        "
+        ---
+
+        ### ⚠ Dependency Lookup Warnings ⚠
+
+        Warnings were logged while processing this repo. Please check logs for more information.
```
Comment (HonkingGoose) [inline lib/workers/repository/errors-warnings.spec.ts:140]:
```suggestion
        Warnings were logged while processing this repo. Please check the logs for more information.
```

File: lib/workers/repository/errors-warnings.ts [hunk 6]
```diff
@@ -69,14 +70,12 @@ export function getDepWarningsPR(
...
     `\n---\n\n### :warning: Dependency Lookup Warnings :warning:\n\n`
   );
-  warningText += `Please correct - or verify that you can safely ignore - these lookup failures before you merge this PR.\n\n`;
-  for (const w of warnings) {
-    warningText += `-   \`${w}\`\n`;
+  warningText += 'Warnings were logged while processing this repo. ';
+  if (dependencyDashboard) {
+    warningText += `Please check the Dependency Dashboard for more information.\n\n`;
+  } else {
+    warningText += `Please check logs for more information.\n\n`;
```
Comment (HonkingGoose) [inline lib/workers/repository/errors-warnings.ts:77]:
```suggestion
    warningText += `Please check the logs for more information.\n\n`;
```

File: lib/workers/repository/update/pr/body/index.spec.ts [hunk 7]
```diff
@@ -199,5 +200,54 @@ describe('workers/repository/update/pr/body/index', () => {
       const match = prDebugDataRe.exec(res);
       expect(match?.groups?.payload).toBeString();
     });
+
+    it('pr body warning', async () => {
+      const massagedMarkDown =
+        '---\n\n### ⚠ Dependency Lookup Warnings ⚠\n\n' +
+        'Warnings were logged while processing this repo. ' +
+        'Please check the Dependency Dashboard for further information\n\n---';
```
Comment (HonkingGoose) [inline lib/workers/repository/update/pr/body/index.spec.ts:208]:
```suggestion
        'Please check the Dependency Dashboard for more information\n\n---';
```

File: lib/workers/repository/update/pr/body/index.spec.ts [hunk 8]
```diff
@@ -199,5 +200,54 @@ describe('workers/repository/update/pr/body/index', () => {
...
+
+      const compiledContent =
+        '---\n\n\n\n### ⚠ Dependency Lookup Warnings ⚠' +
+        '\n\n\n\nWarnings were logged while processing this repo. ' +
+        'Please check the Dependency Dashboard for further information\n\n\n\n---';
```
Comment (HonkingGoose) [inline lib/workers/repository/update/pr/body/index.spec.ts:213]:
```suggestion
        'Please check the Dependency Dashboard for more information\n\n\n\n---';
```

File: lib/workers/repository/update/pr/body/index.spec.ts [hunk 9]
```diff
@@ -199,5 +200,54 @@ describe('workers/repository/update/pr/body/index', () => {
...
+          debugData: {
+            updatedInVer: '1.2.3',
+            createdInVer: '1.2.3',
+          },
+        }
+      );
+      const expected =
+        '---\n\n### ⚠ Dependency Lookup Warnings ⚠' +
+        '\n\nWarnings were logged while processing this repo. ' +
+        'Please check the Dependency Dashboard for further information\n\n---';
```
Comment (HonkingGoose) [inline lib/workers/repository/update/pr/body/index.spec.ts:249]:
```suggestion
        'Please check the Dependency Dashboard for more information\n\n---';
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
Some more fixes to the text.

**Review by HonkingGoose — DISMISSED** [review]

File: lib/workers/repository/errors-warnings.ts [hunk 10]
```diff
@@ -69,14 +70,12 @@ export function getDepWarningsPR(
   warningText = emojify(
     `\n---\n\n### :warning: Dependency Lookup Warnings :warning:\n\n`
   );
-  warningText += `Please correct - or verify that you can safely ignore - these lookup failures before you merge this PR.\n\n`;
```
Comment (rarkins) [inline lib/workers/repository/errors-warnings.ts:72]: Isn't this line from the onboarding PR? Will it be replaced with the new messages? Is this intentional?

Comment (betterPeleg) [reply lib/workers/repository/errors-warnings.ts:72]: we wanted to keep the warning text in the PR body as simple as possible.
do you want to add a "safely ignore" message

**Review by rarkins — CHANGES_REQUESTED** [review]
I think we had a misunderstanding about the purpose of this issue. I have updated it in https://github.com/renovatebot/renovate/issues/16924 so that it is clearer. There should be no impact on the onboarding PR

**Comment by betterPeleg** [comment]
so onboarding should stay as it was before

**Comment by MaronHatoum** [comment]
> I think we had a misunderstanding about the purpose of this issue. I have updated it in #16924 so that it is clearer. There should be no impact on the onboarding PR

**onboarding PR:** https://github.com/MaronHatoum/pr-warnings/pull/2
**dependencyDashboard=true:** https://github.com/MaronHatoum/pr-warnings/pull/6
**dependencyDashboard=false:** https://github.com/MaronHatoum/pr-warnings/pull/5

File: lib/workers/repository/errors-warnings.spec.ts [hunk 11]
```diff
@@ -203,4 +247,60 @@ describe('workers/repository/errors-warnings', () => {
       expect(res).toBe('');
     });
   });
+
+  describe('getDepWarningsOnboardingPR()', () => {
+    beforeEach(() => {
+      jest.resetAllMocks();
+    });
```
Comment (viceice) [inline lib/workers/repository/errors-warnings.spec.ts:254]: remove, you don't use any mocks below

**Review by HonkingGoose — APPROVED** [review]

**Review by HonkingGoose — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**